src/Features/Core/Portable/EmbeddedLanguages/EmbeddedLanguageCommentDetector.cs

```diff
@@ -22,11 +22,29 @@ namespace Microsoft.CodeAnalysis.EmbeddedLanguages
     {
         private readonly Regex _regex;
 
+        private const int MaxRegexCacheSize = 11;
```

Comment:
Honestly, you don't need the max size. Wee own this space and the set is in the single digits.

Reply:
But I would feel all kinds of wrong add an unbounded cache of regexes. This makes me feel somewhat less susceptible to future embarrassment.

Reply:
so it's reaslly not unbounded.  it's exactly the same as the nubmer of embedded languages we support (currently 3 :)).  TBH, i'm not sure why these detectors are recomputed, instead of having a few static instances of htem in a well known location.